alacritty/src/wayland_theme.rs

```diff
@@ -18,9 +18,11 @@ impl AlacrittyWaylandTheme {
         let hovered_close_icon = colors.normal().red;
         let hovered_maximize_icon = colors.normal().green;
         let hovered_minimize_icon = colors.normal().yellow;
-        let foreground = colors.primary.foreground;
-        let background = colors.primary.background;
-        let dim_foreground = colors.primary.dim_foreground.unwrap_or(foreground * DIM_FACTOR);
+        let foreground = colors.search_bar_foreground();
+        let background = colors.search_bar_background();
+        // TODO decide how to dim properly, maybe we can derive it from the foreground color
+        // with our factor and some formula.
+        let dim_foreground = foreground * DIM_FACTOR;
```

Comment:
It'll work with bright foreground color, however not with dark. The intent here is to show color that is less noticeable than the default one, so the formula should be adjusted to account for dark colors properly. Are you fine with computing it like

```rust
if foreground.r + foreground.g + foreground.b < 3 * 128 {
    (255 - (255 - foreground) * DIM_FACTOR)
} else {
    foreground * DIM_FACTOR
}
```

This is pretty arbitrary but should be symmetric at least. We can compute comparison differently, I don't mind.

Reply:
I don't really like conditionals for logic like this. I feel like it would be nicer to have just a quick single line formula that does it. But I'm not sure I can propose one.

In theory what I'd want is to reduce the opacity tbh. So we'd take the foreground color and merge it with the background color at a factor of DIM_FACTOR?

But can't you also set the opacity for each of these anyways, so can we maybe just pass the DIM_FACTOR as opacity for dim_foreground? Well DIM_FACTOR * 256?

Reply:
Oh, yeah, you can just blend them with some factor. I'll look into that.

Reply:
I've decided to blend them with 0.5 factor, so it'll look the same on light/dark background,since this dimming doesn't use dim colors from our config in any way.